scripts/post_autosave.js

```diff
@@ -0,0 +1,164 @@
+
+            // The amount of time (ms) after typing before posts are cached 
+            CACHE_TIME: 500,
+
+            // Local copy of the cache to use within a single page view
+            cache: [],
+
+            timeoutId: -1,
+
+            install: function () {
+                PostAutosave.cache = getSetting(PostAutosave.CACHE_SETTING_NAME);
+
+                var frm_submit = document.getElementById('frm_submit');
+                var notificationArea = document.createElement('div');
+
+                notificationArea.id = "autosaveNotification";
+
+                frm_submit.parentNode.appendChild(notificationArea);
+            },
+
+            installClickEvent: function (postbox) {
+                clearTimeout(PostAutosave.timeoutId);
+                
+                var frm_body = document.getElementById('frm_body');
+                var frm_submit = document.getElementById('frm_submit');
+
+                var id = document.getElementById('parent_id').value;
+
+                frm_body.value = PostAutosave.getAutosave(id);
+                frm_body.addEventListener('input', function () { PostAutosave.triggerAutosave(id); }, true);
+
+                // Removes the cached post after it is submitted
+                frm_submit.addEventListener('click', function () { PostAutosave.removeAutosave(id); });
+            },
+
+            triggerAutosave: function (id) {
+                clearTimeout(PostAutosave.timeoutId);
+                PostAutosave.displayNotification("Saving...", false);
+
+                var frm_body = document.getElementById('frm_body');
+
+                PostAutosave.timeoutId = setTimeout(PostAutosave.setAutosave, PostAutosave.CACHE_TIME, id, frm_body.value);
+            },
+
+            getAutosave: function (id) {
+                clearTimeout(PostAutosave.timeoutId);
+
+                var posts = PostAutosave.cache;
+
+                for (var i = 0; i < posts.length; i++) {
+                    if (posts[i].id == id) {
+                        PostAutosave.displayNotification("Draft restored", false);
+
+                        return posts[i].text;
+                    }
+                }
+
+                return '';
+            },
+
+            setAutosave: function (id, text) {
+                clearTimeout(PostAutosave.timeoutId);
+
+                if (!PostAutosave.updateAutosave(id, text))
+                    PostAutosave.addAutosave(id, text);
+
+                PostAutosave.displayNotification("Draft saved", false);
+            },
+
+            addAutosave: function (id, text) {
+                if (text.length == 0)
+                    return;
+
+                var posts = PostAutosave.cache;
+                var post = { id: id, text: text };
+
+                posts.unshift(post);
+
+                // Truncate the cache if necessary
+                if (posts.length > PostAutosave.CACHE_SIZE)
+                    posts.splice(PostAutosave.CACHE_SIZE);
```

Comment:
If you use my object suggestion, I would simplify the logic of these three methods (set/add/update).

When updating, just set posts[i].text = updatedText instead of reforming the array.